### DISCUSSION TITLE
serve does not reload changes with Docker

### DISCUSSION THREAD

**Discussion by jaceklaskowski** [squidfunk/mkdocs-material#2750] [General]
Hi,

I'm curious if others are experiencing issues with `mkdocs serve` that does not seem to reload changes when used in Docker and a mounted directory.

My setup uses a custom-made Docker image `FROM squidfunk/mkdocs-material-insiders:INSIDERS_TAG` that is executed using:

```
docker run --rm \
  -it \
  -p 8000:8000 \
  -v ${PWD}:/docs \
  jaceklaskowski/mkdocs-material-insiders \
  serve --dirtyreload --verbose --dev-addr 0.0.0.0:8000
```

All files are in https://github.com/japila-books/japila-books.github.io.

Jacek

**Comment by squidfunk** [comment]
Thanks for reporting. I assume it also happens with the none-Insiders version? Could be related to Docker / MkDocs itself.

**Reply by squidfunk** [reply]
Does this first happen with the latest version, or did it never work?

**Reply by jaceklaskowski** [reply]
It works with `mkdocs<1.2` in [requirements.txt](https://github.com/squidfunk/mkdocs-material/blob/master/requirements.txt#L22). Right after I `git restore requirements.txt` and build the image it stops reloading pages (although I can see messages in the log about monitoring files).

I haven't had much time to investigate it further as it works OK locally (with the change) and while deploying to github pages.

**Reply by squidfunk** [reply]
I see, so it's related to MkDocs 1.2 then, which includes a new livereload implementation. Could you open an issue on their issue tracker?

**Reply by squidfunk** [reply]
Related: https://github.com/mkdocs/mkdocs/issues/2457

**Comment by mmacfadden** [comment]
Thanks for this awesome project. Also glad to have this discussion forum here. I just ran into this issue in Docker myself and spent a bit debugging it.  Glad to have found this discussion so I know its not just me.
